### PR TITLE
Add lockout role functionality

### PR DESCRIPTION
**Pull request by UlisseMini**
@Scoder12

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -2,17 +2,40 @@
...
+    while True:
+        await asyncio.sleep(every)
+        res = db.execute(
+            "SELECT id, user_id, role_id FROM timed_roles WHERE role_id = ? AND remove_role_at < ?",
+            (
+                LOCKOUT_ROLE_ID,
+                int(time.time()),
+            ),
+        )
+        for _, user_id, role_id in res:
```
Comment (spencerpogo) [inline src/main.py:26]: why select id just to discard it

Comment (spencerpogo) [reply src/main.py:26]: oh bc you use it below nvm

File: src/main.py [hunk 2]
```diff
@@ -2,17 +2,40 @@
...
+            guild = bot.get_guild(GUILD_ID)
+            user = await guild.fetch_member(user_id)
+            await user.remove_roles(guild.get_role(role_id))
+
+        db.execute("DELETE FROM timed_roles WHERE id IN (" + ','.join('?' for _ in res) + ")", [_id for _id, _ in res])
```
Comment (spencerpogo) [inline src/main.py]: this should be an executemany

File: src/utils.py [hunk 3]
```diff
@@ -33,3 +33,14 @@ def id_from_mention(mention):
...
         raise ValueError(f"Value is not a mention: {mention!r}")
     return int(m.group(1))
+
+
+def parse_duration(dur: str):
+    "Parse a duration of the form 1h 2m 3s into a number of seconds"
+    try:
+        return sum(
+            int(s[:-1]) * {"h": 60 * 60, "m": 60, "s": 1}[s[-1]]
+            for s in dur.lower().split()
```
Comment (spencerpogo) [inline src/utils.py:43]: I don't really like that spaces are mandatory here? but it would add complexity

**Review by spencerpogo — CHANGES_REQUESTED** [review]

File: src/main.py (hunk 2 above)
Comment (UlisseMini) [reply src/main.py]: Fixed ty

**Review by spencerpogo — APPROVED** [review]

**Pull request merged**